sparse-traits/src/spaces/element.rs

```diff
@@ -4,25 +4,25 @@ use super::LinearSpace;
 pub trait Element {
     /// Item type of the vector.
     type Space: LinearSpace;
-    type View<'a>
+    type View<'b>
     where
-        Self: 'a;
-    type ViewMut<'a>
+        Self: 'b;
+    type ViewMut<'b>
     where
-        Self: 'a;
+        Self: 'b;
 
     /// Return the underlying space.
     fn space(&self) -> &Self::Space {
         std::unimplemented!();
     }
 
-    fn view<'a>(&'a self) -> Self::View<'a>;
+    fn view<'b>(&'b self) -> Self::View<'b>;
 
-    fn view_mut<'a>(&'a mut self) -> Self::ViewMut<'a>;
+    fn view_mut<'b>(&'b mut self) -> Self::ViewMut<'b>;
 }
 
 // The view type associated with elements of linear spaces.
-pub type ElementView<'a, Space> = <<Space as LinearSpace>::E as Element>::View<'a>;
+pub type ElementView<'a, 'b, Space> = <<Space as LinearSpace>::E<'a> as Element>::View<'b>;
```

Comment:
If I'm following correctly, the reason for the `'a` is to ensure at compile-time that the space lives at least as long as any of the elements (vectors). An alternative would be to use reference counting. I'm a little fearful of how intrusive the compile-time approach may be, since most users don't want to think carefully about their spaces and lifetimes.

Reply:
Users don't encounter these lifetimes. The example files show how to use the functionality and there is no need to index any lifetime. Implementation was a bit of a pain at first with the hierarchy of lifetimes. But this is clean now and behaves as expected.